fdbserver/RESTKmsConnector.actor.cpp

```diff
@@ -0,0 +1,1143 @@
+			}
+
+			tokensPayloadSize += fileSize;
+			if (tokensPayloadSize > FLOW_KNOBS->REST_KMS_CONNECTOR_VALIDATION_TOKENS_MAX_PAYLOAD_SIZE) {
+				TraceEvent("ValidationToken_PayloadTooLarge", ctx->uid)
+				    .detail("MaxAllowedSize", FLOW_KNOBS->REST_KMS_CONNECTOR_VALIDATION_TOKENS_MAX_PAYLOAD_SIZE);
+				throw value_too_large();
+			}
+
+			// Populate validation token details
+			ValidationTokenCtx tokenCtx = ValidationTokenCtx(tokenName, VALIDATION_TOKEN_SOURCE_FILE);
+			tokenCtx.value.resize(fileSize);
+			ifs.seekg(0, std::ios_base::beg);
+			ifs.read((char*)tokenCtx.value.data(), fileSize);
+			tokenCtx.filePath = tokenFile;
+
+			// NOTE: avoid logging token-value to prevent token leaks in log files..
+			TraceEvent("ValidationToken_ReadFile", ctx->uid)
+			    .detail("TokenName", Traceable<std::string>::toString(tokenCtx.name))
+			    .detail("TokenSize", tokenCtx.value.size())
+			    .detail("TokenFilePath", Traceable<std::string>::toString(tokenCtx.filePath.get()))
+			    .detail("TotalPayloadSize", tokensPayloadSize);
+
+			ctx->validationTokens.emplace(tokenName, std::move(tokenCtx));
+
+			ifs.close();
+		} catch (Error& e) {
+			if (ifs.is_open()) {
+				ifs.close();
+			}
+			throw e;
+		}
+	}
+}
+
+void procureValidationTokens(Reference<RESTKmsConnectorCtx> ctx) {
+	const std::string& mode = FLOW_KNOBS->REST_KMS_CONNECTOR_VALIDATION_TOKEN_MODE;
+
+	if (mode.compare("file") == 0) {
+		procureValidationTokensFromFiles(ctx, StringRef(FLOW_KNOBS->REST_KMS_CONNECTOR_VALIDATION_TOKEN_FILE_DETAILS));
+	} else {
+		throw not_implemented();
+	}
+}
+
+ACTOR Future<Void> connectorCore_impl(KmsConnectorInterface interf) {
+	state Reference<RESTKmsConnectorCtx> self = makeReference<RESTKmsConnectorCtx>(interf.id());
+
+	TraceEvent("RESTKmsConnector_Init", self->uid).log();
+
+	wait(discoverKmsUrls(self));
+	procureValidationTokens(self);
+
+	loop {
+		choose {
+			when(KmsConnLookupEKsByKeyIdsReq req = waitNext(interf.ekLookupByIds.getFuture())) {
+				state KmsConnLookupEKsByKeyIdsReq byKeyIdReq = req;
+				state KmsConnLookupEKsByKeyIdsRep byKeyIdResp;
+				try {
+					KmsConnLookupEKsByKeyIdsRep _rByKeyId = wait(fetchEncryptionKeyByKeyId(self, byKeyIdReq));
+					byKeyIdResp = _rByKeyId;
+					byKeyIdReq.reply.send(byKeyIdResp);
+				} catch (Error& e) {
+					TraceEvent("LookupEKsByKeyIds_Failed", self->uid).error(e);
+					byKeyIdReq.reply.sendError(e);
+				}
+			}
+			when(KmsConnLookupEKsByDomainIdsReq req = waitNext(interf.ekLookupByDomainIds.getFuture())) {
+				state KmsConnLookupEKsByDomainIdsReq byDomainIdReq = req;
+				state KmsConnLookupEKsByDomainIdsRep byDomainIdResp;
+				try {
+					KmsConnLookupEKsByDomainIdsRep _rByDomainId =
+					    wait(fetchEncryptionKeyByDomainId(self, byDomainIdReq));
+					byDomainIdResp = _rByDomainId;
+					byDomainIdReq.reply.send(byDomainIdResp);
+				} catch (Error& e) {
+					TraceEvent("LookupEKsByDomainIds_Failed", self->uid).error(e);
+					byDomainIdReq.reply.sendError(e);
```

Comment:
We likely want to be smart about error handling, and not try to send actor_cancelled over the network, for example.

Reply:
Good point. I'll create list of acceptable error that are worth passing upstream. Thanks.